src/modalities/models/gpt2/gpt2_model.py

```diff
@@ -831,22 +854,30 @@ def __init__(
                             attention_config=attention_config,
                             dropout=dropout,
                             ffn_hidden=ffn_hidden,
-                            attention_norm=deepcopy(attention_norm),
-                            ffn_norm=deepcopy(ffn_norm),
+                            # deepcopy did not work here! The weights were then automatically
+                            # moved to a cuda device even when the deepcopied weights were on
+                            # a meta device!
+                            attention_norm=attention_norm_config.norm_type.value(**dict(attention_norm_config.config)),
```

Comment:
Here, we could use a factory.

Outside of GPT2LLM:

def make_norm(norm_config):
    norm_cls = norm_config.norm_type.value
    return lambda: norm_cls(**norm_config.config)

attention_norm_factory = make_norm(attention_norm_config)
ffn_norm_factory = make_norm(ffn_norm_config)

Inside GPT2LLM:

h=nn.ModuleList(
    [
        GPT2Block(
            ...,
            attention_norm=attention_norm_factory(),
            ffn_norm=ffn_norm_factory(),
        )
        for _ in range(n_layer)
    ]
)

Reply:
Should we integrate that at a later step, maybe? 
I tried to make a minimal change w.r.t. norm instantiation.

Reply:
Sounds good - Please create an issue.